akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -1292,13 +1292,61 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
    *
    * '''Cancels when''' downstream cancels
    *
-   * @param seed Provides the first state for extrapolation using the first unconsumed element
    * @param extrapolate Takes the current extrapolation state to produce an output element and the next extrapolation
    *                    state.
    */
   def expand[U](extrapolate: function.Function[Out, java.util.Iterator[U]]): javadsl.Flow[In, U, Mat] =
     new Flow(delegate.expand(in ⇒ extrapolate(in).asScala))
 
+  /**
+   * Allows a faster downstream to progress of a slower upstream.
+   *
+   * This is achieved by introducing "extrapolated" elements - based on those from upstream - whenever downstream
+   * signals demand.
+   *
+   * Pressurize does not support [[akka.stream.Supervision#restart]] and [[akka.stream.Supervision#resume]].
+   * Exceptions from the `extrapolate` function will complete the stream with failure.
+   *
+   * '''Emits when''' downstream stops backpressuring, AND EITHER upstream emits OR initial element is present OR
+   * `extrapolate` is non-empty and applicable
+   *
+   * '''Backpressures when''' downstream backpressures or current `extrapolate` runs empty
+   *
+   * '''Completes when''' upstream completes and current `extrapolate` runs empty
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   * @param extrapolate takes the current upstream element and provides a sequence of "extrapolated" elements based
+   *                    on the original, to be emitted in case downstream signals demand
+   */
+  def pressurize[U >: Out](extrapolate: function.Function[U, java.util.Iterator[U]]): javadsl.Flow[In, U, Mat] =
```

Comment:
Naming is hard, sure but we can stick to simply `extrapolate`, since:

> good name

yup

>    but but too phonetically close to expand to not be confusing

Does not matter? I mean, it's a close operation anyway, so that's good.

>    on top of that, expand already has an argument extrapolate (which as far as I understand can't be renamed without breaking binary compatibility, correct me if I'm wrong on this)

Does not really matter I think. Parameter names can be changed in a bin compat way btw, it's doable via `@deprecatedName`, but I don't think there's a need to do this. Pressurize sounds very weird as the only thing about pressure we call such is backpressure, and while this one does relate to it it's a bit of a weird name ;)

Reply:
In fairness, "backpressure" is probably also a new concept for the majority of Akka Streams newcomers, hence the suggestion. But I won't argue - `extrapolate` is the better choice obviously :).

Re bin compat - what I had in mind is a potential problem with named argument assignments. But on second thought, that should be transformed by compiler into positional arguments anyway - and, from the emitted bytecode, I see that's the case, at least in Scala 2.12.

Reply:
> Re bin compat - what I had in mind is a potential problem with named argument assignments. 

As I said, `@deprecatedName` fixes this

Reply:
Just realized I left this thread dangling in the wind, sorry. Yup, you're right of course that `@deprecatedName` would be a way to handle the concern (thanks BTW for pointing this out, I learned something new).

However, if I understand correctly, you don't see a point in going through the whole deprecation procedure in this case, right?